docs/src/api/customize.md

```diff
@@ -0,0 +1,111 @@
+
+| Field | Description |
+| ----- | ----------- |
+| `controller` | The whole CompositeController object, like what you might get from `kubectl get compositecontroller <name> -o json`. |
+| `parent` | The parent object, like what you might get from `kubectl get <parent-resource> <parent-name> -o json`. |
+
+## Customize Hook Response
+
+The body of your response should be a JSON object with the following fields:
+
+| Field | Description |
+| ----- | ----------- |
+| `relatedResources` | A list of JSON objects representing all the desired related resource label selectors. |
+
+The `relatedResources` field should contain a flat list of objects,
+not an associative array.
+
+Each resource rule object should be a JSON object with the following fields:
+
+| Field | Description |
+| ----- | ----------- |
+| `apiVersion` | The API `<group>/<version>` of the parent resource, or just `<version>` for core APIs. (e.g. `v1`, `apps/v1`, `batch/v1`) |
+| `resource`   | The canonical, lowercase, plural name of the parent resource. (e.g. `deployments`, `replicasets`, `statefulsets`) |
+| `labelSelector` | A `v1.LabelSelector` object. **Do not left it nil** - see note below |
+| `namespace` | Optional. The Namespace to select in |
+| `names` | Optional. A list of strings, representing individual objects to return |
+
+Field `namespaces` and `names` are optional. If present, first label selector is applied, then abovementioned fields.
+
+**Important note**
+If you do not want match related resources based on `labelSelector`
+(i.e. select all resources of given type), then left it empty, (`'labelSelector': {}`) as not 
+setting this field will cause it to be set to `nil`, thus matching no objects.
+
+If the parent resource is cluster scoped and the related resource is namespaced,
+the namespace may be used to restrict which objects to look at. If the parent
+resource is namespaced, the related resources must come from the same namespace.
+Specifying the namespace is optional, but if specified must match.
+
+Note that your webhook handler must return a response with a status code of `200`
+to be considered successful. Metacontroller will wait for a response for up to the
+amount defined in the [Webhook spec](../api/hook.md#webhook).
+
+## Example
+
+Let's take a look at [Global Config Map example](../examples.md#global-config-map) custom resource object:
+```yaml
+---
+apiVersion: examples.metacontroller.io/v1alpha1
+kind: GlobalConfigMap
+metadata:
+  name: globalsettings
+spec:
+  sourceName: globalsettings
+  sourceNamespace: global
+```
+it tells that we would like to have `globalsettings` ConfigMap from `global` namespace
+present in each namespace.
+
+The customize hook request will looks like :
+```json
+{
+    'controller': '...',
+    'parent': '...'
+}
+```
+
+and we need to extract information identyfying source ConfigMap.
+
+We return :
+```json
+[
+    {
+        'apiVersion': 'v1',
+        'resource': 'configmaps',
+        'labelSelector': {},
```

Comment:
This bit here confuses me - `'labelSelector': {} means - select all objects` - doesn't that mean this section should select all objects?

If we're ignoring it because `names` is specified - then we should be able to leave it blank.

Needing to specify it, but having it have no effect is odd, and we should try to avoid that

Reply:
So, first label selector is applied, then names/namespace.

So in case we want to ignore label selector and do not filter any object by it, it must select all objects.
Then filter by namespace and names are applied.


In general logic is (https://github.com/grzesuav/metacontroller/blob/related/controller/common/customize/manager.go#L262)
logic is as follows :
- apply label selector
- apply namesace
- apply names

So in case you want to match just using namespace/names, you must put label selecto returning all objects. to not filter out any object basing on labels

Reply:
this is the part I changed to match k8s api logic. when we leave it empty selector is deserialized as `nil`, and is matching `no objects` as all.

Reply:
I can adjust logic to select all objects when `labelSelector` is missing if it will be more obvious for users

Reply:
adjusted logic and documentation